components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineMiddleware.kt

```diff
@@ -104,7 +104,14 @@ private fun createEngineSession(
         return null
     }
 
-    val engineSession = engine.createSession(tab.content.private, tab.contextId)
+    val manifestDisplayMode = session.webAppManifest?.display
+
+    val engineSession = if (manifestDisplayMode != null) {
+        engine.createSessionWithDisplayMode(tab.content.private, manifestDisplayMode, tab.contextId)
+    } else {
+        engine.createSession(tab.content.private, tab.contextId)
+    }
+
```

Comment:
The `browser-session` component no longer exists and this part would need to be updated.